gm4_block_compressors/data/gm4_block_compressors/functions/machine/create.mcfunction

```diff
@@ -0,0 +1,18 @@
+# places the block_compressor down
+# @s = player who placed the block_compressor
+# located at the center of the placed block
+# run from gm4_block_compressors:machine/verify_place_down
+
+# place dropper
+setblock ~ ~ ~ dropper[facing=down]{CustomName:'{"translate":"%1$s%3427655$s","with":[{"translate":"%1$s%3427656$s","with":[{"text":"Compression Amount","font":"minecraft:default","color":"#373737"},[{"text":"Compression\\u00a0Amount","font":"gm4:half_scale"},{"text":"Compression\\u00a0Amount","font":"gm4:inverted"},{"text":"Compression\\u00a0Amount","font":"gm4:inverted_spacing"},{"text":"Compression\\u00a0Amount","font":"gm4:offscreen"},{"translate":"gui.gm4.block_compressor","font":"gm4:container_gui","color":"white"},{"text":"Compression\\u00a0Amount","font":"gm4:half_scale"},{"text":"Compression\\u00a0Amount","font":"gm4:inverted"},{"text":"Compression\\u00a0Amount","font":"gm4:inverted_spacing"},{"text":"Compression Amount","font":"minecraft:default","color":"#373737"}]]},{"translate":"%1$s%3427656$s","with":[{"translate":"container.gm4.block_compressor","font":"minecraft:default","color":"#373737"},[{"translate":"container.gm4.block_compressor","font":"gm4:half_scale"},{"translate":"container.gm4.block_compressor","font":"gm4:inverted"},{"translate":"container.gm4.block_compressor","font":"gm4:inverted_spacing"},{"translate":"container.gm4.block_compressor","font":"gm4:offscreen"},{"translate":"gui.gm4.block_compressor","font":"gm4:container_gui","color":"white"},{"translate":"container.gm4.block_compressor","font":"gm4:half_scale"},{"translate":"container.gm4.block_compressor","font":"gm4:inverted"},{"translate":"container.gm4.block_compressor","font":"gm4:inverted_spacing"},{"translate":"container.gm4.block_compressor","font":"minecraft:default","color":"#373737"}]]}]}'}
```

Comment:
Beeps, would you mind explaining briefly how the custom-font stuff here works? From the laymans perspective it seems like theres a lot of redundant data here.

Reply:
The redundant data is there in order to ensure the name and GUI remains centered. Basically the way the custom font stuff is working in this instance is that it's printing the name with half the normal width, then prints it in reverse then prints the name. That all happens with an offset so the player doesn't see it. Then it prints the string that actually shows the GUI texture. Then it once again prints the name with half width and then in reverse. Then it prints the actual name of the container that the players see. Then it does that all again with the custom namespace which means the translation shows up instead of the English set name. 

TL;DR the reason it prints the name a bunch of times is to keep it centered. If you look at one of the hopper names (e.g. ender hoppers) you will see that it is much simpler.